Modules/main.c

```diff
@@ -1381,7 +1383,8 @@ pymain_read_conf(_PyMain *pymain, _PyCoreConfig *config,
         config->coerce_c_locale = new_coerce_c_locale;
 
         /* The encoding changed: read again the configuration
-           with the new encoding */
+           with the new encoding, but free allocated memory beforehand: */
+        pymain_clear_pymain(pymain);
```

Comment:
Please move the call after pymain_clear_cmdline() call a few lines before, and leave the comment unchanged.

Reply:
Ok, done